packages/api/app/Middleware/RegistrationCompleted.js

```diff
@@ -2,22 +2,46 @@
 /** @typedef {import('@adonisjs/framework/src/Response')} Response */
 /** @typedef {import('@adonisjs/framework/src/View')} View */
 
-const RegistrationUncompletedException = use('App/Exceptions/RegistrationUncompletedException');
+const { errors, errorPayload } = require('../Utils');
 
 class RegistrationCompleted {
-	async handle({ auth }, next, properties) {
+	async handle({ auth, request, response }, next, properties) {
 		const user = await auth.getUser();
 
-		if (properties[0] === 'be_curator') {
-			if (!user.toJSON().can_be_curator) {
-				throw new RegistrationUncompletedException();
-			}
-		}
+		const unCompletedFields = [];
+
+		await Promise.all(
+			properties.map(async (check) => {
+				if (check === 'check_personal_data') {
+					const unCompletedFieldsPD = user.getCheckPersonalData();
+					if (unCompletedFieldsPD.length) {
+						unCompletedFields.push(...unCompletedFieldsPD);
+					}
+				}
+				if (check === 'check_academic_data') {
+					const unCompletedFieldsAD = await user.getCheckAcademicData();
+					if (unCompletedFieldsAD.length) {
+						unCompletedFields.push(...unCompletedFieldsAD);
+					}
+				}
+				if (check === 'check_organizational_data') {
+					const unCompletedFieldsOD = await user.getCheckOrganizationalData();
+					if (unCompletedFieldsOD.length) {
+						unCompletedFields.push(...unCompletedFieldsOD);
+					}
+				}
+			}),
+		);
```

Comment:
I think it will work well 🤔
```suggestion
		const mappingCheckMethods = {
			check_personal_data: user.getCheckPersonalData,
			check_academic_data: user.getCheckAcademicData,
			check_organizational_data: user.getCheckOrganizationalData,
		};

		const unCompletedFields = await Promise.all(
			properties.map(async (check) => mappingCheckMethods[check]() || null).filter(Boolean),
		);
```